tests/python/pants_test/task_test_base.py

```diff
@@ -328,13 +336,6 @@ def _create_task(self, task_type, context):
     """Helper method to instantiate tasks besides self._testing_task_type in the test workdir."""
     return task_type(context, self.test_workdir)
 
-  @dataclass(frozen=True)
```

Comment:
Why move this into the global namespace? I think it was intentionally an inner class.

Reply:
Because I didn't manage to properly qualify it for the return type of the function while it was within the scope (see commit message for a bit more details)

Reply:
BTW, if you know a way, I would of course prefer that.

Reply:
Was the issue line 339 with the type hint? If so, put it in quotes like `DeclarativeTaskTestMixin.TaskInvocationResult`. Type hints can't "forward reference" things that have not been properly declared yet. For example, this won't work:

```python
class Foo:
  def factory() -> Foo:
    return cls()
```

It must instead be:
```python
class Foo:
  def factory() -> 'Foo':
    return cls()
```
Python 3.7 introduced `from __future__ import annotations` to fix this.